editor/themes/editor_theme_manager.cpp

```diff
@@ -0,0 +1,2232 @@
+	p_theme->set_color("font_color",                      "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/text_color"));
+	p_theme->set_color("line_number_color",               "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/line_number_color"));
+	p_theme->set_color("caret_color",                     "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/caret_color"));
+	p_theme->set_color("font_selected_color",             "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/text_selected_color"));
+	p_theme->set_color("selection_color",                 "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/selection_color"));
+	p_theme->set_color("brace_mismatch_color",            "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/brace_mismatch_color"));
+	p_theme->set_color("current_line_color",              "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/current_line_color"));
+	p_theme->set_color("line_length_guideline_color",     "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/line_length_guideline_color"));
+	p_theme->set_color("word_highlighted_color",          "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/word_highlighted_color"));
+	p_theme->set_color("bookmark_color",                  "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/bookmark_color"));
+	p_theme->set_color("breakpoint_color",                "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/breakpoint_color"));
+	p_theme->set_color("executing_line_color",            "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/executing_line_color"));
+	p_theme->set_color("code_folding_color",              "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/code_folding_color"));
+	p_theme->set_color("folded_code_region_color",        "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/folded_code_region_color"));
+	p_theme->set_color("search_result_color",             "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/search_result_color"));
+	p_theme->set_color("search_result_border_color",      "CodeEdit", EDITOR_GET("text_editor/theme/highlighting/search_result_border_color"));
+	/* clang-format on */
+}
+
+// Public interface for theme generation.
+
+Ref<Theme> EditorThemeManager::generate_theme(const Ref<Theme> &p_old_theme) {
+	OS::get_singleton()->benchmark_begin_measure("EditorTheme", "Generate Theme");
+
+	Ref<Theme> theme = _create_base_theme(p_old_theme);
+
+	const String custom_theme_path = EDITOR_GET("interface/theme/custom_theme");
+	if (!custom_theme_path.is_empty()) {
+		Ref<Theme> custom_theme = ResourceLoader::load(custom_theme_path);
+		if (custom_theme.is_valid()) {
+			theme->merge_with(custom_theme);
+		}
+	}
+
+	OS::get_singleton()->benchmark_end_measure("EditorTheme", "Generate Theme");
+	return theme;
+}
+
+bool EditorThemeManager::must_regenerate_theme() {
```

Comment:
Is this method used to check if a theme regeneration is needed?

If so, I would name it something less ambiguous, like `is_theme_regeneration_needed()`. At first glance, I thought that `must_regenerate_theme()` had side-effects.

Reply:
Renamed it to `is_generated_theme_outdated()` to be as explicit as possible.